### ISSUE TITLE
"Zip file closed" when using Class.forName(String)

### ISSUE THREAD

**Issue by mfnalex** [Technove/AirplaneLite#33]
Hi, I'm the author of [AngelChest ](https://github.com/JEFF-Media-GbR/Spigot-AngelChestPlus)

One of my users keeps getting the following stacktrace when using my plugin:

```
[04:05:10] [Server thread/ERROR]: Could not pass event PlayerDeathEvent to AngelChest v3.29.1
java.lang.IllegalStateException: zip file closed
	at java.util.zip.ZipFile.ensureOpen(ZipFile.java:828) ~[?:?]
	at java.util.zip.ZipFile.getEntry(ZipFile.java:327) ~[?:?]
	at java.util.jar.JarFile.getEntry(JarFile.java:513) ~[?:?]
	at java.util.jar.JarFile.getJarEntry(JarFile.java:468) ~[?:?]
	at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:168) ~[patched_1.16.5.jar:git-Airplane-9]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:586) ~[?:?]
	at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:119) ~[patched_1.16.5.jar:git-Airplane-9]
	at org.bukkit.plugin.java.JavaPluginLoader.getClassByName(JavaPluginLoader.java:235) ~[patched_1.16.5.jar:git-Airplane-9]
	at org.bukkit.plugin.java.PluginClassLoader.loadClass0(PluginClassLoader.java:132) ~[patched_1.16.5.jar:git-Airplane-9]
	at org.bukkit.plugin.java.PluginClassLoader.loadClass(PluginClassLoader.java:114) ~[patched_1.16.5.jar:git-Airplane-9]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:519) ~[?:?]
	at java.lang.Class.forName0(Native Method) ~[?:?]
	at java.lang.Class.forName(Class.java:375) ~[?:?]
	at de.jeff_media.angelchest.hooks.GenericHooks.isSlimefunSoulbound(GenericHooks.java:139) ~[?:?]
	at de.jeff_media.angelchest.hooks.GenericHooks.keepOnDeath(GenericHooks.java:149) ~[?:?]
	at de.jeff_media.angelchest.data.AngelChest.removeKeptItems(AngelChest.java:609) ~[?:?]
	at de.jeff_media.angelchest.data.AngelChest.<init>(AngelChest.java:294) ~[?:?]
	at de.jeff_media.angelchest.listeners.PlayerListener.spawnAngelChest(PlayerListener.java:618) ~[?:?]
	at de.jeff_media.angelchest.listeners.PlayerListener.spawnAngelChestHighest(PlayerListener.java:746) ~[?:?]
	at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor714.execute(Unknown Source) ~[?:?]
	at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69) ~[patched_1.16.5.jar:git-Airplane-9]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:76) ~[patched_1.16.5.jar:git-Airplane-9]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[patched_1.16.5.jar:git-Airplane-9]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:624) ~[patched_1.16.5.jar:git-Airplane-9]
	at org.bukkit.craftbukkit.v1_16_R3.event.CraftEventFactory.callPlayerDeathEvent(CraftEventFactory.java:876) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.EntityPlayer.die(EntityPlayer.java:997) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.EntityLiving.damageEntity(EntityLiving.java:1365) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.EntityHuman.damageEntity(EntityHuman.java:877) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.EntityPlayer.damageEntity(EntityPlayer.java:1161) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.EntityArrow.a(EntityArrow.java:371) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.IProjectile.a(IProjectile.java:150) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.IProjectile.preOnHit(IProjectile.java:141) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.EntityArrow.tick(EntityArrow.java:214) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.EntityTippedArrow.tick(EntityTippedArrow.java:108) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.WorldServer.entityJoinedWorld(WorldServer.java:1502) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.WorldServer.doTick(WorldServer.java:1132) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.MinecraftServer.b(MinecraftServer.java:1552) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.DedicatedServer.b(DedicatedServer.java:439) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.MinecraftServer.a(MinecraftServer.java:1402) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1134) ~[patched_1.16.5.jar:git-Airplane-9]
	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:292) ~[patched_1.16.5.jar:git-Airplane-9]
	at java.lang.Thread.run(Thread.java:831) [?:?]
```

The first part where my plugin is mentioned in the stacktrace is [THIS](https://github.com/JEFF-Media-GbR/Spigot-AngelChestPlus/blob/76a63c3ce3759f259f479357c7d5a2bc09550847/src/main/java/de/jeff_media/angelchest/hooks/GenericHooks.java#L139). It's just a simple Class.forName(String) inside a try/catch block.

The user who reported this is running OpenJDK 16.0.1+8 on Ubuntu 18.0.4.

It seems like something's wrong with the ClassLoader or something? They sent me their AngelChest.jar and I checked that it's not corrupted.

Anyone got any idea on why it happens? Noone else reported this problem yet, I am a bit confused. It's running fine on Spigot and Paper.

**Comment by mfnalex** [comment]
Just in case this might be important: The user does NOT have Slimefun installed (which would provide the Class I am trying to access with Class.forName)

EDIT: /reload, PlugMan or ServerUtils etc. are not involved, the plugin also enabled properly.

**Comment by PaulBGD** [comment]
Pushed maybe a fix for 1.17, I'll port it back to 1.16 when I have some time. It's not inherently an issue with Airplane, but it doesn't seem like any of the other forks are maintaining 1.16 right now so I'll try to get it fixed here.

**Comment by mfnalex** [comment]
Thank you!
